### PR TITLE
Ignore RCS1043 for WPF classes that inherit from FrameworkElement

### LINKED ISSUE #1174: partial keyword usage and WPF
In WPF a control can be composed of an xaml file and an xaml.cs file.

In this case it has to have the keyword partial because the xaml file is compiled to code.

But roslynator doesn't seem to take this into account and tries to remove the keyword partial from classes that inherit from UserControl.

### PR DESCRIPTION
**Pull request by bdovaz**
Fix #1174

Analyzing WPF frameworks like Prism and reading the official documentation, it seems that taking into account the `FrameworkElement` class is enough. If it is not, it will be easy to modify it in the future.

References:

https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/base-elements

https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/base-elements-overview

https://github.com/PrismLibrary/Prism/tree/master/src/Wpf/Prism.Wpf

### CODE REVIEW

**Comment by bdovaz** [comment]
@JosefPihrt can you merge and release another version pease?

Thanks!

File: src/Analyzers/CSharp/Analysis/RemovePartialModifierFromTypeWithSinglePartAnalyzer.cs [hunk 1]
```diff
@@ -12,7 +13,13 @@ namespace Roslynator.CSharp.Analysis;
 public sealed class RemovePartialModifierFromTypeWithSinglePartAnalyzer : BaseDiagnosticAnalyzer
 {
     private static ImmutableArray<DiagnosticDescriptor> _supportedDiagnostics;
-    private static readonly MetadataName _componentBaseName = MetadataName.Parse("Microsoft.AspNetCore.Components.ComponentBase");
+
+    private static readonly MetadataName[] _componentBaseNames = new MetadataName[] {
```
Comment (JosefPihrt) [inline src/Analyzers/CSharp/Analysis/RemovePartialModifierFromTypeWithSinglePartAnalyzer.cs:17]: Can you change field name to something else? _metadataNames would be probably fine.

**Review by JosefPihrt — CHANGES_REQUESTED** [review]

File: src/Analyzers/CSharp/Analysis/RemovePartialModifierFromTypeWithSinglePartAnalyzer.cs (hunk 1 above)
Comment (bdovaz) [reply src/Analyzers/CSharp/Analysis/RemovePartialModifierFromTypeWithSinglePartAnalyzer.cs]: Done!

**Comment by JosefPihrt** [comment]
@bdovaz Please update changelog.

**Comment by bdovaz** [comment]
> @bdovaz Please update changelog.

Done!

File: ChangeLog.md [hunk 2]
```diff
@@ -10,28 +10,29 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Fixed
 
-- Fix [RCS1187](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1187.md) ([#1150](https://github.com/JosefPihrt/Roslynator/pull/1150)).
-- Fix [RCS1056](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1056.md) ([#1154](https://github.com/JosefPihrt/Roslynator/pull/1154)).
-- Fix [RCS1208](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1208.md) ([#1153](https://github.com/JosefPihrt/Roslynator/pull/1153))
+- Fix [RCS1187](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1187) ([#1150](https://github.com/JosefPihrt/Roslynator/pull/1150)).
+- Fix [RCS1056](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1056) ([#1154](https://github.com/JosefPihrt/Roslynator/pull/1154)).
+- Fix [RCS1208](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1208) ([#1153](https://github.com/JosefPihrt/Roslynator/pull/1153)).
+- Fix [RCS1043](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1043) ([#1176](https://github.com/JosefPihrt/Roslynator/pull/1153)).
```
Comment (JosefPihrt) [inline ChangeLog.md:35]:
```suggestion
- Fix [RCS1043](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1043) ([#1176](https://github.com/JosefPihrt/Roslynator/pull/1176)).
```

**Review by JosefPihrt — APPROVED** [review]

**Comment by JosefPihrt** [comment]
Thanks for the PR!

**Pull request merged**